tests/nataliia_romanchuk/homework_4/test_collections.py

```diff
@@ -0,0 +1,424 @@
+some_str = "abcd"
+some_tuple = "a", "b", "c", "d"
+some_list = ["a", "b", "c", "d"]
+some_set = {"a", "b", "c", "d"}
+some_dict = {"a": 123, "b": 456, "c": 789, "d": 100}
+
+
+# len
+def test_len_string():
+    assert len(some_str) == 4
+
+
+def test_len_tuple():
+    assert len(some_tuple) == 4
+
+
+def test_len_list():
+    assert len(some_list) == 4
+
+
+def test_len_set():
+    assert len(some_set) == 4
+
+
+def test_len_dict():
+    assert len(some_dict) == 4
+
+
+# contains
+def test_in_string():
+    assert "a" in some_str
+
+
+def test_in_tuple():
+    assert "a" in some_tuple
+
+
+def test_in_list():
+    assert "a" in some_list
+
+
+def test_in_set():
+    assert "a" in some_set
+
+
+def test_in_dict_keys():
+    assert "a" in some_dict
+
+
+def test_in_dict_values():
+    assert "a" in some_dict.keys()
+    assert "a" not in some_dict.values()
+
+
+# getitem
+def test_indexing_string():
+    assert some_str[0] == "a"
+
+
+def test_indexing_tuple():
+    assert some_tuple[0] == "a"
+
+
+def test_indexing_list():
+    assert some_list[0] == "a"
+
+
+def test_indexing_dict():
+    assert some_dict["a"] == 123
+
+
+# last item
+def test_last_item_string():
+    assert some_str[-1] == "d"
+
+
+def test_last_item_tuple():
+    assert some_tuple[-1] == "d"
+
+
+def test_last_item_list():
+    assert some_list[-1] == "d"
+
+
+# modify
+
+
+def test_modify_and_delete_list():
+    a = [1, 2, 3, 4]
+    a[3] = "abc"
+    assert a[3] == "abc"
+
+
+def test_delete_from_list():
+    a = [1, 2, 3, 4]
+    del a[1]
+    assert a == [1, 3, 4]
+
+
+def test_modify_dict():
+    b = {"a": 123, "b": 456}
+    b["c"] = 7889
+    assert b["c"] == 7889
+
+
+def test_modify_dict():
+    b = {"a": 123, "b": 456}
+    b["b"] = "some"
+    assert b["b"] == "some"
+
+
+def test_delete_from_dict():
+    b = {"a": 123, "b": 456}
+    del b["a"]
+    assert "a" not in b
+
+
+# unpack / iterate
+def test_list_conversion():
+    assert list(some_str) == ['a', 'b', 'c', 'd']
+    assert list(some_tuple) == ['a', 'b', 'c', 'd']
+    assert list(some_list) == ['a', 'b', 'c', 'd']
+    assert sorted(list(some_set)) == ['a', 'b', 'c', 'd']
+    assert list(some_dict) == ['a', 'b', 'c', 'd']
+
+
+def test_tuple_conversion():
+    assert tuple(some_str) == ('a', 'b', 'c', 'd')
+    assert tuple(some_tuple) == ('a', 'b', 'c', 'd')
+    assert tuple(some_list) == ('a', 'b', 'c', 'd')
+    assert tuple(some_set) == ('a', 'b', 'c', 'd')
+    assert tuple(some_dict) == ('a', 'b', 'c', 'd')
```

Comment:
for set and dict - no ordering guarantes !!!

Reply:
fixed